bf: S3C-3829 Ignore placeholder log entries

naren-rajendran opened the pull request:
Notification populator should not process placeholder log entries created for deletion or cleanup.

bert-e commented:
# Hello naren-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3829-ignore-placeholder-entries` into
`development/7.9`
* [w/8.1/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.1` (pull request #1347)
* [w/8.2/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.2` (pull request #1348)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

miniscruff commented:
Bugfix without a test change?

anurag4DSB commented:
> Bugfix without a test change?

I thought it will go to integration, right?

naren-rajendran commented:
> > Bugfix without a test change?
> 
> I thought it will go to integration, right?

Yes.

rahulreddy commented:
> > > Bugfix without a test change?
> > 
> > 
> > I thought it will go to integration, right?
> 
> Yes.

There should be a test here for this (unit/functional/both), we can't wait for a Integration test to give us feedback if this is fixed or not. I understand having a test setup here is probably hard but we should not use that as an excuse to defer testing in Integration.

Commit:
bf: S3C-3829 Ignore placeholder log entries

bert-e commented:
# History mismatch



Merge commit #e351359a496c41c22b026861aee5916f0670dbad on the integration branch
`w/8.1/bugfix/S3C-3829-ignore-placeholder-entries` is merging a branch which is neither the current
branch `bugfix/S3C-3829-ignore-placeholder-entries` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3829-ignore-placeholder-entries` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3829-ignore-placeholder-entries` into
`development/7.9`
* [w/8.1/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.1` (pull request #1352)
* [w/8.2/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.2` (pull request #1353)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

bert-e commented:
# History mismatch



Merge commit #30b73d456f2fb270e91b95eec82dbe7b2fd55e6e on the integration branch
`w/8.1/bugfix/S3C-3829-ignore-placeholder-entries` is merging a branch which is neither the current
branch `bugfix/S3C-3829-ignore-placeholder-entries` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3829-ignore-placeholder-entries` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
> > > > Bugfix without a test change?
> > > 
> > > 
> > > I thought it will go to integration, right?
> > 
> > 
> > Yes.
> 
> There should be a test here for this (unit/functional/both), we can't wait for a Integration test to give us feedback if this is fixed or not. I understand having a test setup here is probably hard but we should not use that as an excuse to defer testing in Integration.

I refactored backbeat to handle this in util function and added a test to it.

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3829-ignore-placeholder-entries` into
`development/7.9`
* [w/8.1/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.1` (pull request #1354)
* [w/8.2/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.2` (pull request #1355)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

miniscruff commented on extensions/notification/utils/config.js:
```diff
@@ -69,6 +69,14 @@ function filterConfigsByEvent(bnConfigs, event) {
 function validateEntry(bnConfig, entry) {
     const { bucket, eventType } = entry;
+    /**
+     * if the event type is undefined, it is most likely an entry that is a
+     * placeholder for deletion or cleanup, these entries should be ignored and
+     * not be processed.
```
* You say "is undefined" but use a falsey check below. Should this be `eventType === undefined`?
* "most likely" is not a very concrete description, it sounds like there is a chance it's something else. What that else is I have no idea.

naren-rajendran replied:
corrected.

bert-e commented:
# History mismatch



Merge commit #68a919522c805dd6c8b234c2b2d5666dd4912970 on the integration branch
`w/8.1/bugfix/S3C-3829-ignore-placeholder-entries` is merging a branch which is neither the current
branch `bugfix/S3C-3829-ignore-placeholder-entries` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3829-ignore-placeholder-entries` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3829-ignore-placeholder-entries` into
`development/7.9`
* [w/8.1/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.1` (pull request #1356)
* [w/8.2/bugfix/S3C-3829-ignore-placeholder-entries](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3829-ignore-placeholder-entries) will be merged into `development/8.2` (pull request #1357)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

naren-rajendran commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue S3C-3829.

Goodbye naren-scality.
